cms/static/js/views/utils/create_library_utils.js

```diff
@@ -1,58 +1,17 @@
 /**
  * Provides utilities for validating libraries during creation.
  */
-define(["jquery", "underscore", "gettext", "js/views/utils/view_utils"],
-    function ($, _, gettext, ViewUtils) {
+define(["jquery", "gettext", "js/views/utils/view_utils", "js/views/utils/create_utils_base"],
+    function ($, gettext, ViewUtils, CreateUtilsFactory) {
         "use strict";
         return function (selectors, classes) {
-            var validateTotalKeyLength, setNewLibraryFieldInErr, hasInvalidRequiredFields,
-                createLibrary, validateFilledFields, configureHandlers;
+            var keyLengthViolationMessage = gettext("The combined length of the organization and library code fields cannot be more than <%=limit%> characters.")
+            var keyFieldSelectors = [selectors.org, selectors.number];
+            var nonEmptyCheckFieldSelectors = [selectors.name, selectors.org, selectors.number];
 
-            var validateRequiredField = ViewUtils.validateRequiredField;
-            var validateURLItemEncoding = ViewUtils.validateURLItemEncoding;
+            CreateUtilsFactory.call(this, selectors, classes, keyLengthViolationMessage, keyFieldSelectors, nonEmptyCheckFieldSelectors);
```

Comment:
`.call` is a function that's available on JavaScript function objects. It's a version of the function that allows you to call a function while specifying an alternative `this`. So when we call the helper function via `.call`, anything that is actually set on `this` in the helper function is set on the function that called it.

This link goes over `this`, from the basics to more involved uses like in a `.call`. It may aid in understanding to read the whole thing, since it builds up to the explanation:

http://www.w3schools.com/js/js_function_invocation.asp